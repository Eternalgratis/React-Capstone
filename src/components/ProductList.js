import { useDispatch, useSelector } from 'react-redux';
import { useEffect, useState } from 'react';
import { fetchProducts } from '../redux/productApi';
import FinalProducts from './finalProduct';
import './HomePage.css';

const ProductList = () => {
  const [searchTerm, setSearchTerm] = useState('');
  const dispatch = useDispatch();
  const { products } = useSelector((state) => state.products);
  const filteredProducts = products.filter((product) => product.title.toLowerCase()
    .includes(searchTerm.toLowerCase()));
  useEffect(() => {
    dispatch(fetchProducts());
  }, [searchTerm]);

  const handleChange = (e) => {
    const { value } = e.target;
    setSearchTerm(value);
  };

  return (
    <div className="list-container">
      <div className="input-container">
        <input
          type="text"
          name="search"
          placeholder="Search Item"
          className="search"
          value={searchTerm}
          onChange={handleChange}
        />
      </div>

      <div className="allList-container">

        {filteredProducts.map((product) => (
          <FinalProducts
            id={product.id}
            key={product.id}
            title={product.title}
            price={product.price}
            description={product.description}
            image={product.image}
          />
        ))}
      </div>
    </div>
  );
};

export default ProductList;
