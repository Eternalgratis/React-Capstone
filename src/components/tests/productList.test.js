import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import ProductList from '../ProductList';
import store from '../../redux/configureStore';

describe('to test the productList component', () => {
  test('matches productList DOM snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <ProductList />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
