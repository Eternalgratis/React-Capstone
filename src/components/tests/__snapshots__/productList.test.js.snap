// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to test the productList component matches productList DOM snapshot 1`] = `
<div
  className="list-container"
>
  <div
    className="input-container"
  >
    <input
      className="search"
      name="search"
      onChange={[Function]}
      placeholder="Search Item"
      type="text"
      value=""
    />
  </div>
  <div
    className="allList-container"
  />
</div>
`;
